app/editor/src/features/admin/ingests/configurations/Newspaper.tsx

```diff
@@ -26,6 +27,50 @@ export const Newspaper: React.FC = (props) => {
   const sources = lookups.sources.map((s) => new OptionItem(s.name, s.code, s.isEnabled));
   const source = sources.find((t) => t.value === values.configuration.defaultSource);
 
+  const [items, setItems] = React.useState<any[]>([]);
+
+  React.useEffect(() => {
+    const data: any[] = [];
+    let id = 0;
+    const keyValues = values.configuration.sources.split('&');
+    keyValues.forEach((x: string) => {
+      if (x !== '=') data.push({ id: ++id, name: x.split('=')[0], source: x.split('=')[1] });
+    });
+    data.push({ id: ++id, name: '', source: '' });
+    setItems(data);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
If you're only using this to initialize the state of `items` you can create a function and place it in the `React.useState( ... )`

Reply:
Updated, thanks